File: CCC-school-work/comporg/pnc2/bschulte/go/primeregbs.go
package main

import "fmt"
import "os"
import "time"
import "math"
import "strconv"

func main() {

	var err     error
	var upBound int
	upBound, err = strconv.Atoi(os.Args[1])
	if err != nil{
		fmt.Print(err)
	}

	start := time.Now()

	num   := 2
	i     := 3
	divBy := 2

	fmt.Fprintf(os.Stderr, "2 3 ")


	for num!=upBound{

		mod := i%divBy
		sqt := math.Sqrt(float64(i))

		if mod!=0{
			if divBy!=int(sqt){
				divBy+=1
			} else if divBy == int(sqt){
				fmt.Fprintf(os.Stderr, "%d ", i)
				i+=1
				num+=1
				divBy=2
			}else{
				fmt.Fprintf(os.Stderr, "%d ", i)
				i+=1
				num+=1
				divBy=2
			}
		} else{
			i+=1
			divBy=2
		}

	}


	end := time.Now()


	elapsed := end.Sub(start)
	tm := float64(elapsed) / float64(time.Second)

	fmt.Fprintf(os.Stdout, "%8.4f\n", tm)
}

File: CCC-school-work/comporg/pnc2/bschulte/lua/primeregbs.lua
require "os"
require "io"
require "string"

local divBy, i, num = 2, 3, 2
local upBound = tonumber(arg[1])

start = os.clock()

io.stderr:write("2 3 ")

while num ~= upBound do
	local sqt = math.sqrt(i)
	local mod = i%divBy
	sqt=math.floor(sqt)
	if mod ~= 0 then
		if divBy ~= sqt then
			divBy=divBy+1
		elseif divBy==sqt then
			io.stderr:write(i)
			io.stderr:write(" ")
			i=i+1
			num=num+1
			divBy=2
		end
	else
		i=i+1
		divBy=2
	end
end


stop = os.clock()

io.write(string.format("\n%8.4f\n", stop - start))
